snapcraft/internal/parts.py

```diff
@@ -270,7 +270,8 @@ def load_plugin(self, part_name, plugin_name, part_properties):
             plugin_name=plugin_name,
             part_properties=part_properties,
             project_options=self._project_options,
-            part_schema=self._validator.part_schema)
+            part_schema=self._validator.part_schema,
+            definitions_schema=self._validator.definitions_schema)
```

Comment:
Unfortunately jsonschema explodes with a `RecursionError` when the YAML contains recursive anchors, so it seems that using jsonschema refs appears to be our only option.